### PR TITLE
Refactor Appointment class. Add testing for NullAppointment

### PR DESCRIPTION
**Pull request by Kb-Tay**
Refactor Appointment class to adhere to LSP. 

Abstract class ScheduleItem created that is the parent class of Appointment and NullAppointment.

This is done so that NullAppointment and Appointment do not break expectations of a scheduleItem.

### CODE REVIEW

File: src/main/java/seedu/address/model/appointment/NullAppointment.java [hunk 1]
```diff
@@ -3,10 +3,15 @@
...
+public class NullAppointment extends ScheduleItem implements Comparable<ScheduleItem> {
     public static final String MESSAGE_NULL_APT = "No Appointment made!";
-    public NullAppointment() {
-        super("", null);
+    private static final NullAppointment nullappointment = new NullAppointment();
+    private final String value;
+    private NullAppointment() {
+        this.value = MESSAGE_NULL_APT;
+    }
+    public static NullAppointment getNullappointment() {
```
Comment (AlyssaPng) [inline src/main/java/seedu/address/model/appointment/NullAppointment.java:13]: i think can rename this to getNull Appointment

File: src/main/java/seedu/address/model/appointment/Appointment.java [hunk 2]
```diff
@@ -120,10 +120,11 @@ public static Appointment parseAppointmentDescription(String appointment) {
...
     @Override
-    public int compareTo(Appointment appointment) {
-        if (appointment instanceof NullAppointment) {
+    public int compareTo(ScheduleItem scheduleItem) {
+        if (scheduleItem instanceof NullAppointment) {
             return 0;
         } else {
+            Appointment appointment = (Appointment) scheduleItem;
             return this.date.compareTo(appointment.date);
         }
```
Comment (AlyssaPng) [inline src/main/java/seedu/address/model/appointment/Appointment.java:129]: good to check if it is Appointment before type-casting

**Review by AlyssaPng — COMMENTED** [review]
LGTM. Just some minor naming conventions issues.

**Review by nikele2001 — APPROVED** [review]
Looks good. Perhaps a suggestion would be to use camelcasing for the method names. Otherwise, good job! Perhaps one thing that we can consider is to make the association between Person and Appointment classes to be bidirectional, so that we can know who the appointment is for from the Appointment object itself.

**Pull request merged**